src/libs/Url.js

```diff
@@ -10,7 +12,64 @@ function addTrailingForwardSlash(url) {
     return url;
 }
 
+/**
+ * Parse href to URL object
+ * @param {String} href
+ * @returns {Object}
+ */
+function getURLObject(href) {
+    const urlRegex = new RegExp(URL_WEBSITE_REGEX, 'gi');
+    const match = urlRegex.exec(href);
+    if (!match) {
+        return {
+            href: undefined,
+            protocol: undefined,
+            hostname: undefined,
+            path: undefined,
+        };
+    }
+    const baseUrl = match[0];
+    const protocol = match[1];
+    return {
+        href,
+        protocol,
+        hostname: baseUrl.replace(protocol, ''),
+        path: href.startsWith(baseUrl) ? href.replace(baseUrl, '') : '',
+    };
+}
+
+/**
+ * Determine if we should remove w3 from hostname
+ * @param {String} hostname
+ * @returns {Boolean}
+ */
+function shouldRemoveW3FromExpensifyUrl(hostname) {
```

Comment:
Ah, also this function is not working as expected, locally. I mean for internal engineers as our local URL has subdomain and can be accessed with both www and now www - expensify.com.dev

Reply:
I'm so appreciated if we had the regex for this case before. If not I'll find by myself. Thanks

Reply:
I am really confused and not sure how can we fix this problem once and for all. Should we keep a list of URLs that support `www` and adjust this function? 

@MonilBhavsar I think we need internal support here.

Reply:
I feel like this is an exceptional case and we can add a condition to support this.

Reply:
@MonilBhavsar You mean if `hostname === 'www.expensify.com.dev'` we will remove `www.` first then keep the logic to remove www as before right?

Reply:
Correct!
Also please leave a comment, why are we doing like this. Something like - 
> Since expensify.com.dev is accessible with and without www subdomain.